src/main/java/com/bootcamp/demo/service/ScooterService.java

```diff
@@ -0,0 +1,86 @@
+package com.bootcamp.demo.service;
+
+import com.bootcamp.demo.model.Scooter;
+import com.bootcamp.demo.service.exception.ServiceException;
+import com.google.api.core.ApiFuture;
+import com.google.cloud.firestore.DocumentSnapshot;
+import com.google.cloud.firestore.Firestore;
+import com.google.cloud.firestore.QueryDocumentSnapshot;
+import com.google.cloud.firestore.WriteResult;
+import org.apache.commons.lang3.exception.ExceptionUtils;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+import org.springframework.stereotype.Service;
+
+import java.util.Set;
+import java.util.concurrent.ExecutionException;
+import java.util.stream.Collectors;
+import java.util.stream.StreamSupport;
+
+@Service
+public class ScooterService {
+
+    private static final Logger LOGGER = LoggerFactory.getLogger(ScooterService.class);
+    private static final String COLLECTION_SCOOTERS_PATH = "bookings/databases/scooters";
+    private final Firestore db;
+
+    private ScooterService(Firestore db) {
+        this.db = db;
+    }
+
+    public Scooter findScooterById(String scooterId) {
+        LOGGER.info("FIND SCOOTER BY ID - service function invoked: scooterId = {}", scooterId);
```

Comment:
Notes about logging: https://sematext.com/blog/java-logging-best-practices/ . You can find more details here regarding best practices for logging, including recommended approach for content.

Reply:
Usually, when possible, it is good to log the information that uniquely identifies the entry. For example, if you would have an exception on scooter's retrieval, it would be helpful to have the id in the log, since then you could look in the db and check if anything was changed to its structure in a way that prevents deserialization. Also, this is not sensitive information.
If it would be for a user on the other hand, it would be preferable to log the id and not the email, since the later one would be covered by GDPR. As a note, if you would need to log such information (sensitive), you should convert it in such a way that it cannot be converted back to the original value (consider here why for example hashing could be used and encryption not; you can find an overview of these approaches here: https://www.geeksforgeeks.org/difference-between-hashing-and-encryption/)
In conclusion, it is ok to use the id.

Reply:
Ok. Thanks for the detailed answer/suggestions.